monai/data/meta_tensor.py

```diff
@@ -0,0 +1,148 @@
+        Else, use the default value. Similar for the affine, except this could come from
+        four places.
+        Priority: `affine`, `meta["affine"]`, `x.affine`, `get_default_affine`.
+        """
+        out: MetaTensor = torch.as_tensor(x, *args, **kwargs).as_subclass(cls)  # type: ignore
+        # set meta
+        if meta is not None:
+            out.meta = meta
+        elif isinstance(x, MetaObj):
+            out.meta = x.meta
+        else:
+            out.meta = out.get_default_meta()
+        # set the affine
+        if affine is not None:
+            if "affine" in out.meta:
+                warnings.warn("Setting affine, but the applied meta contains an affine. " "This will be overwritten.")
+            out.affine = affine
+        elif "affine" in out.meta:
+            pass  # nothing to do
+        elif isinstance(x, MetaTensor):
+            out.affine = x.affine
+        else:
+            out.affine = out.get_default_affine()
+        out.affine = out.affine.to(out.device)
+
+        return out
+
+    def _copy_attr(self, attribute: str, input_objs: list[MetaObj], default_fn: Callable, deep_copy: bool) -> None:
+        super()._copy_attr(attribute, input_objs, default_fn, deep_copy)
+        val = getattr(self, attribute)
+        if isinstance(val, torch.Tensor):
+            setattr(self, attribute, val.to(self.device))
+
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None) -> torch.Tensor:
+        """Wraps all torch functions."""
+        if kwargs is None:
+            kwargs = {}
+        ret: MetaTensor = super().__torch_function__(func, types, args, kwargs)
+        # e.g., __repr__ returns a string
+        if not isinstance(ret, torch.Tensor):
+            return ret
+        if not (get_track_meta() or get_track_transforms()):
+            return ret.as_tensor()
+        meta_args = MetaObj.flatten_meta_objs(list(args) + list(kwargs.values()))
+        ret._copy_meta(meta_args)
+        ret.affine = ret.affine.to(ret.device)
+        return ret
+
+    def get_default_affine(self) -> torch.Tensor:
+        return torch.eye(4, device=self.device)
+
+    def as_tensor(self) -> torch.Tensor:
+        """
+        Return the `MetaTensor` as a `torch.Tensor`.
+        It is OS dependent as to whether this will be a deep copy or not.
+        """
+        return self.as_subclass(torch.Tensor)  # type: ignore
+
+    def as_dict(self, key: str) -> dict:
+        """
+        Get the object as a dictionary for backwards compatibility.
+
+        Args:
+            key: Base key to store main data. The key for the metadata will be
+                determined using `PostFix.meta`.
+
+        Return:
+            A dictionary consisting of two keys, the main data (stored under `key`) and
+                the metadata.
+        """
+        return {key: self.as_tensor(), PostFix.meta(key): self.meta}
+
+    @property
+    def affine(self) -> torch.Tensor:
```

Comment:
Hi @rijobro ,

May I know why change `affine` property from `MetaObj` to `MetaTensor`? Seems it's not `Tensor` specific logic?
I didn't read all the comments and conversations, maybe I missed some points.

Thanks in advance.